lib/openfoodfacts.dart

```diff
@@ -131,6 +132,37 @@ class OpenFoodAPIClient {
     return Status.fromApiResponse(response.body);
   }
 
+  /// Add the given product to the database.
+  /// Returns a Status object as result.
+  static Future<Status> addProductFields(
+    final User user,
+    final String barcode,
+    final Map<AddableProductField, String> map, {
+    QueryType? queryType,
+  }) async {
+    if (map.isEmpty) {
+      throw Exception('Empty map');
```

Comment:
You could, maybe, have a more relevant message like "Please add at least one AddableProductField"